internal/access/organization.go

```diff
@@ -62,3 +66,16 @@ func DeleteOrganization(c *gin.Context, id uid.ID) error {
 func SanitizedDomain(subDomain, serverBaseDomain string) string {
 	return strings.ToLower(subDomain) + "." + serverBaseDomain
 }
+
+// DomainAvailable is needed to check if an org domain is available before completing social sign-up
+func DomainAvailable(c *gin.Context, domain string) error {
+	rCtx := GetRequestContext(c)
+	_, err := data.GetOrganization(rCtx.DBTxn, data.GetOrganizationOptions{ByDomain: domain})
+	if err != nil {
+		if !errors.Is(err, internal.ErrNotFound) {
+			logging.L.Error().Err(err).Msg("failed to check if organization exists by domain")
```

Comment:
Might be safer to return an error here since we don't actually know if the domain is in use.

Reply:
Updated this to fail